data_sources/gateway_polling.go

```diff
@@ -39,25 +45,32 @@ func StartGatewayPolling(conf config.GatewayPolling, measurements chan<- parser.
 	})
 	log.Info("Starting gateway polling")
 	stop := make(chan bool)
-	go gatewayPoller(conf.GatewayUrl, interval, measurements, stop, log)
+	go gatewayPoller(conf.GatewayUrl, conf.BearerToken, interval, measurements, stop, log)
 	return stop
 }
 
-func gatewayPoller(url string, interval time.Duration, measurements chan<- parser.Measurement, stop <-chan bool, log *log.Entry) {
+func gatewayPoller(url string, bearer_token string, interval time.Duration, measurements chan<- parser.Measurement, stop <-chan bool, log *log.Entry) {
 	seenTags := make(map[string]int64)
-	poll(url, measurements, seenTags, log)
+	poll(url, bearer_token, measurements, seenTags, log)
 	for {
 		select {
 		case <-stop:
 			return
 		case <-time.After(interval):
-			poll(url, measurements, seenTags, log)
+			poll(url, bearer_token, measurements, seenTags, log)
 		}
 	}
 }
 
-func poll(url string, measurements chan<- parser.Measurement, seenTags map[string]int64, log *log.Entry) {
-	resp, err := http.Get(url + "/history")
+func poll(url string, bearer_token string, measurements chan<- parser.Measurement, seenTags map[string]int64, log *log.Entry) {
+	req, err := http.NewRequest("GET", url+"/history", nil)
```

Comment:
This `err` value is never used, either replace it with with the blank identifier (`_`) or add a check for the (very unlikely) error

Reply:
Added error handling.